binary-compatibility-validator/test/PublicApiTest.kt

```diff
@@ -68,3 +72,26 @@ class PublicApiTest(
             error("No single file matching $regex in $libsDir:\n${files.joinToString("\n")}")
     }
 }
+
+private val ATOMIC_FU_REF = "Lkotlinx/atomicfu/".toByteArray()
+
+private fun JarFile.checkForAtomicFu() {
+    var found = false
+    for (e in entries()) {
+        if (e.name.endsWith(".class")) {
+            getInputStream(e).use {
+                val bytes = it.readBytes()
+                loop@for (i in 0 until bytes.size - ATOMIC_FU_REF.size) {
+                    for (j in 0 until ATOMIC_FU_REF.size) {
+                        if (bytes[i + j] != ATOMIC_FU_REF[j]) continue@loop
+                    }
+                    // found!
+                    println("Found atomicfu reference in ${e.name}")
+                    found = true
```

Comment:
Please raise error right here and add classname to the error message

Reply:
It's hard to debug when it fails on the first class, but I'm rewriting it to raise one exception with a list of all affected classes instead of printing the list to stdout.